tests/cache/test-vary-cache.php

```diff
@@ -391,22 +422,58 @@ public function test__validate_cookie_values_invalid( $value, $expected_error_co
 		$actual_result = $get_validate_cookie_value_method->invokeArgs(null, [
 			$value
 		] );
+
 		$this->assertWPError( $actual_result, 'Not WP_Error object' );
 
 		$actual_error_code = $actual_result->get_error_code();
 		$this->assertEquals( $expected_error_code, $actual_error_code, 'Incorrect error code' );
 	}
 
 	public function test__validate_cookie_value_valid( ) {
-
 		$get_validate_cookie_value_method = self::get_method( 'validate_cookie_value' );
 
 		$actual_result = $get_validate_cookie_value_method->invokeArgs(null, [
 			'dev-group'
 		] );
+
 		$this->assertTrue( $actual_result );
+	}
 
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__send_vary_headers__sent_for_group() {
+		Vary_Cache::register_group( 'dev-group' );
+
+		do_action( 'send_headers' );
+
+		$this->assertContains( 'Vary: X-VIP-Go-Segmentation', xdebug_get_headers() );
 	}
 
+	/**
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test__send_vary_headers__sent_for_group_with_encryption() {
+		define( 'VIP_GO_AUTH_COOKIE_KEY', 'abc' );
+		define( 'VIP_GO_AUTH_COOKIE_IV', '123' );
+		Vary_Cache::register_group( 'dev-group' );
+		Vary_Cache::enable_encryption();
+
+		do_action( 'send_headers' );
+
+		$this->assertContains( 'Vary: X-VIP-Go-Auth', xdebug_get_headers() );
```

Comment:
:+1:

Reply:
The xdebug method is a super weird quirk when dealing with headers in phpunit.

Reply:
yeah, it immediately jumped out at me but is cool to know. TIL.